docs/Middleware-Guide.md

```diff
@@ -0,0 +1,82 @@
+# Introduction
+In designing EigenLayer, we aspired to make minimal assumptions about the structure of middlewares built on top of it. If you are getting started looking at the EigenLayer codebase, the Slasher contract contains most of the logic that actually mediates the interactions between EigenLayer and middlewares. Additionally, there is a general-purpose /middleware/ folder, which contains code that can be extended, used directly, or consulted as a reference in building middleware on top of EigenLayer.
+
+## Important Terminology
+### Tasks
+A task in EigenLayer is the discretized unit of work that operators commit to doing when serving a middleware.  
+### Strategies
+A strategy in EigenLayer, its most simple form, is a contract that stakers can use to deposit and delegate their assets to middlewares on EigenLayer. 
+### Quorums
+Each BLSRegistry defines one or two “quorums”; each operator for the middleware may have stake in EigenLayer that falls into either (or both) quorum(s). Each quorum is essentially defined by two vectors: a vector of “Strategies” of interest (in practice this ends up being tokens of interest) and a vector of “weights” or “multipliers”, which define whether certain strategies are weighed more heavily than others within the quorum (e.g. if the middleware desires to give 2x power to a specific token over another token). In the contract code these vectors are condensed into a single array of `StrategyAndWeightingMultiplier` structs. The ServiceManager.owner() has the ability to edit these arrays at will.
+
+
+# Assumptions Make About Middlewares
+1. *Discretization of Services into "Tasks"*: EigenLayer assumes that services manage tasks. In other words, it is assumed that services discretize commitments undertaken by operators, with each task defining the time period for which the service's operators' stakes are placed "at stake", i.e. potentially subject to slashing. Examples of tasks could include a “DataStore” in EigenDA, or posting a state root of another blockchain for a bridge service, etc. Think: what do I want ETH validators to sign off on?
+
+2. *Stake is “At Stake” on Tasks for a Finite Duration*: It is assumed that every task (eventually) resolves. Each operator places their stake in EigenLayer “at stake” on the tasks that they perform. In order to “release” the stake (e.g. so the operator can withdraw their funds), these tasks need to eventually resolve. It is recommended, but not required that tasks have a known duration when they are instantiated.
```

Comment:
add a linebreak after each (currently italicized) main item. this formatting is way too blocky

Reply:
also make RECOMMENDED uppercase, to match RFC 2119 standards (see, e.g. https://eips.ethereum.org/EIPS/eip-1155 for an example of this in an EIP)

Reply:
and perhaps add the standard RFC 2119 statement defining the terms like “MUST”, “MUST NOT”, “REQUIRED”

Reply:
[a2347b9](https://github.com/Layr-Labs/eigenlayer-contracts/pull/16/commits/a2347b9cb6b764b26258fb2e6bdb4fdf0b895acc)